### ISSUE TITLE
Update Profile requires email or phone and removes First and Last Name

### ISSUE THREAD

**Issue by hecrogon** [klaviyo/klaviyo-api-python#71]
Hello,
I would like to comment some issues that I may have found trying to use the python SDK to push `Custom properties`. The payload is quite simple and the only meaningful data that contain are the Id, and the properties: 
```
profile_data = {
    "data": {
        "type": "profile",
        "id": "<ID>",
        "attributes": {
            "properties": { "field7": "Value7" }
        }
    }
}
```

But I have a few problems that don't happen to me if I use the HTTP API directly.

- If I don't assign the `type` doing this:
```
profile_data['data']['type'] = ProfileEnum.PROFILE  
```
I get this pydantic error:
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for ProfilePartialUpdateQueryResourceObject
type
  Input should be 'profile' [type=enum, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.9/v/enum
```

before doing this:
``` 
ProfilePartialUpdateQuery.from_dict(profile_data)
```

- If the payload does not contain either the email or the phone, I get this response:
```
HTTP response body: errors=[GetAccounts4XXResponseErrorsInner(id='26b32ff6-3def-4189-8618-c584914188be', code='invalid', title='Invalid input.', detail='One or more identifiers is required', source=GetAccounts4XXResponseErrorsInnerSource(pointer='/data/attributes', parameter=None))]
```

- Also after including the `email` or the `phone_number` so the update operation gets to push the data, the First Name and the Last Name are removed even when the payload does not contain those keys. AFAIK, that only should happen if they have the None value assigned (not sure if this happens for more attributes).

Thank you for your help!

**Comment by cpcurtis1218** [comment]
Hi there, thank you writing in.  We're currently working on addressing the pydantic error you've shared here related to the profile data type.  For the identifiers, yes the profile needs to be identified by either the email or phone number - this is not related to the SDK but to the requirements of the API itself. For the last part, could you share an example payload?  For this part, it may be best to write into the Support team so that they can take a look at the specific requests and assist.  In addition, please refer to the API developer logs for your account at https://www.klaviyo.com/developer-tools/dashboard as this can be helpful in troubleshooting specific API requests.

**Comment by hecrogon** [comment]
Hello Chuck, thank you for your response,
This is interesting,

"the profile needs to be identified by either the email or phone number"
I mean that using the Python SDK, I needed the Id + (email or phone_number).  I understand that only the Id should be enough as I'm able to update the profile with a HTTP PATCH request . Also in the endpoint documentation, I don't see that the Id + an email or a phone are required.

https://developers.klaviyo.com/en/reference/update_profile

And example of the payload is:
```
profile_data = {
    "data": {
        "type": "profile",
        "id": "01J0Y7K....................",
        "attributes": {
            "properties": { "field": "Value" }
        }
    }
}
```

Basically I only want to add a custom property "field". I open the issue here since I can update  the profile sending that payload if I directly use the HTTP API, for example with python requests. So I understand that the problem is with the Python SDK.

Agan, thank you so much for your help

**Comment by ben-horgan-klaviyo** [comment]
Hi @hecrogon, I think the source of those issues may be that you not passing a dictionary as the request payload (see the note about this in our [Readme](https://github.com/klaviyo/klaviyo-api-python?tab=readme-ov-file#important-notes)). If you just pass `profile_data` without first calling `ProfilePartialUpdateQuery.from_dict`, do you still have these issues?
